log4j-java-async-logger/README.adoc

```diff
@@ -0,0 +1,229 @@
+        </Console>
+        <RollingRandomAccessFile name="LogToRollingRandomAccessFile" fileName="logs/app.log"
+                                 filePattern="logs/$${date:yyyy-MM}/app-%d{MM-dd-yyyy}-%i.log">
+            <PatternLayout>
+                <Pattern>%d %p %c{1.} [%t] %m%n</Pattern>
+            </PatternLayout>
+            <Policies>
+                <TimeBasedTriggeringPolicy/>
+                <SizeBasedTriggeringPolicy size="3 MB"/>
+            </Policies>
+            <DefaultRolloverStrategy max="5"/>
+        </RollingRandomAccessFile>
+
+    </Appenders>
+    <Loggers>
+
+        <!--  asynchronous loggers -->
+       <AsyncLogger  name="org.apache.logging.log4j.samples.async" level="debug">
+            <AppenderRef ref="LogToRollingRandomAccessFile"/>
+        </AsyncLogger>
+        <Root level="debug">
+            <AppenderRef ref="LogToConsole"/>
+        </Root>
+
+    </Loggers>
+</Configuration>
+```
+*We can change the status to `trace`, `debug`, `info`, `warn`,  and `fatal` to enable the internal Log4j events.*
+
+### Configure `Log4J2AsyncLogger.java` Class :
+Now create a logger class `Log4J2AsyncLogger.java` that uses the Log4J2 API to log the messages.
+
+```java
+package org.apache.logging.log4j.async;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class Log4J2AsyncLogger {
+
+    private static final Logger logger = LogManager.getLogger(Log4J2AsyncLogger.class);
+
+    public static void main(String[] args) {
+
+        Log4J2AsyncLogger myLog = new Log4J2AsyncLogger();
+        myLog.getLog("Log4j2 Log");
+
+    }
+
+    private void getLog(String param){
+
+        logger.info("This is a info log");
+
+        // Previously, need to check the log level log to increase performance
+        if(logger.isDebugEnabled()){
+            logger.debug("This is debug log with param : " + param);
+        }
+
+        if(logger.isWarnEnabled()){
+            logger.info("This is warn log with param : " + param);
+        }
+
+        // In Java 8, No need to check the log level, we can do this
+        while (true) //for test rolling file
+            logger.debug("Hello print {}", () -> getValue());
+    }
+
+    static String getValue() {
+        return "Debug Log";
+    }
+}
+```
+
+### Configure `Log4J2AsyncLoggerException.java` Class :
+Create another logger class `Log4J2AsyncLoggerException.java` to create and check exceptions.
+
+```java
+package org.apache.logging.log4j.async;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class Log4J2AsyncLoggerException {
+
+    private static final Logger logger = LogManager.getLogger(Log4J2AsyncLoggerException.class);
+
+    public static void main(String[] args) {
+
+        try {
+            System.out.println(getException());
+        } catch (IllegalArgumentException e) {
+            logger.error("{}", e);
```

Comment:
```suggestion
            logger.error("An error occurred.", e);
```

Logging events have a text component and a throwable component. You should not pass throwables as text, because:

 * some information is lost in the conversion,
 * the user of the application might want to disable the printing of stack traces.

Reply:
> ** [ppkarwasz](/ppkarwasz)**



> Logging events have a text component and a throwable component. You should not pass throwables as text, because:
> 
> * some information is lost in the conversion,
> * the user of the application might want to disable the printing of stack traces.

You are Right. Thanks for the update :).